alica_engine/src/engine/teammanager/TeamManager.cpp

```diff
@@ -82,28 +82,32 @@ void TeamManager::setTeamTimeout(AlicaTime t)
     }
 }
 
-void TeamManager::readSelfFromConfig()
+void TeamManager::readSelfFromConfig(AgentIDConstPtr agentID)
 {
     essentials::SystemConfig& sc = essentials::SystemConfig::getInstance();
     const std::string localAgentName = _engine->getRobotName();
 
-    constexpr auto notAValidID = std::numeric_limits<uint64_t>::max();
-    uint64_t id = sc["Local"]->tryGet<uint64_t>(notAValidID, "Local", "ID", NULL);
-    if (id != notAValidID) {
-        _localAnnouncement.senderID = _engine->getId(id);
-    } else {
-        _localAnnouncement.senderID = _engine->generateId(DEFAULT_AGENT_ID_SIZE);
-        ALICA_DEBUG_MSG("tm: Auto generated id " << _localAnnouncement.senderID);
-        bool persistId = sc["Alica"]->tryGet<bool>(false, "Alica", "PersistID", NULL);
-        if (persistId) {
-            try{
-                auto* configLocal = sc["Local"];
-                configLocal->setCreateIfNotExistent(static_cast<uint64_t>(*_localAnnouncement.senderID), "Local", "ID", NULL);
-                configLocal->store();
-            } catch(...) {
-                ALICA_ERROR_MSG("tm: impossible to store ID " << _localAnnouncement.senderID);
+    if (agentID == nullptr) {
+        constexpr auto notAValidID = std::numeric_limits<uint64_t>::max();
+        uint64_t id = sc["Local"]->tryGet<uint64_t>(notAValidID, "Local", "ID", NULL);
+        if (id != notAValidID) {
+            _localAnnouncement.senderID = _engine->getId(id);
+        } else {
+            _localAnnouncement.senderID = _engine->generateId(DEFAULT_AGENT_ID_SIZE);
+            ALICA_DEBUG_MSG("tm: Auto generated id " << _localAnnouncement.senderID);
+            bool persistId = sc["Alica"]->tryGet<bool>(false, "Alica", "PersistID", NULL);
+            if (persistId) {
+                try{
+                    auto* configLocal = sc["Local"];
+                    configLocal->setCreateIfNotExistent(static_cast<uint64_t>(*_localAnnouncement.senderID), "Local", "ID", NULL);
+                    configLocal->store();
+                } catch(...) {
+                    ALICA_ERROR_MSG("tm: impossible to store ID " << _localAnnouncement.senderID);
+                }
             }
         }
+    } else {
+        _localAnnouncement.senderID = agentID;
```

Comment:
The engine's getId() & _generateId() methods both internally store the ID's in the AgentIDManager but neither is called in the else condition. The AgentIDManager currently does not seem to use the stored ID's but it is better to maintain consistent behaviour.